src/createDomEvent.ts

```diff
@@ -146,7 +144,8 @@ function createEvent(eventParams: EventParams) {
   const { eventProperties, meta, eventType } = getEventProperties(eventParams)
 
   // user defined eventInterface
-  const metaEventInterface = window[meta.eventInterface]
+  // @ts-expect-error
+  const metaEventInterface: Event = window[meta.eventInterface]
```

Comment:
it says window is expecting a number index....

Reply:
The type from `getEventProperties` is correctly inferred, when I check it is the expected type, it's the `window` that can't be indexed. it has the type [`declare var window: Window & typeof globalThis;`](https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L19683)
We can maybe add to that window object...?

Reply:
Hmm strange. maybe if you cast `meta.eventInterface` as `keyof Window`?

Reply:
Unfortunately that doesn't work either.
<img width="410" alt="Screenshot 2021-04-27 at 12 33 56" src="https://user-images.githubusercontent.com/41805560/116234705-d8b89900-a754-11eb-85b8-cfe317d3e5d1.png">

What's interesting that it reckons `meta.eventInterface` is a `string` as opposed to `string | EventInterface` like in the `DomEvent` interface

Reply:
What about `window[meta.eventInterface as keyof Window]`? I think that should work. If it does not I'll take a look myself.

Reply:
That does! I think it's only marginally better than `// @ts-expect-error` though

Reply:
Cool! `ts-expect-error` will blindly swallow any error, so it should only be used as a last resort.